File: code/java/StreetAccountNews/v0/src/main/java/com/factset/sdk/StreetAccountNews/api/StreetAccountHistoricalStoriesApi.java
package com.factset.sdk.StreetAccountNews.api;

import com.factset.sdk.StreetAccountNews.ApiException;
import com.factset.sdk.StreetAccountNews.ApiClient;
import com.factset.sdk.StreetAccountNews.ApiResponse;
import com.factset.sdk.StreetAccountNews.Configuration;
import com.factset.sdk.StreetAccountNews.Pair;

import javax.ws.rs.core.GenericType;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

import com.factset.sdk.StreetAccountNews.models.CheckstatusResponse;
import com.factset.sdk.StreetAccountNews.models.GetfilesResponse;
import java.time.OffsetDateTime;
import com.factset.sdk.StreetAccountNews.models.RequestfilesResponse;
import com.factset.sdk.StreetAccountNews.models.StreetAccountStatus;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen")
public class StreetAccountHistoricalStoriesApi {
  private ApiClient apiClient;

  public StreetAccountHistoricalStoriesApi() {
    this(Configuration.getDefaultApiClient());
  }

  public StreetAccountHistoricalStoriesApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }
  
  private static final Map<Integer, GenericType> streetaccountHistoricalCheckStatusGetResponseTypeMap = new HashMap<Integer, GenericType>();
  static {
    streetaccountHistoricalCheckStatusGetResponseTypeMap.put(200, new GenericType<CheckstatusResponse>(){});
    streetaccountHistoricalCheckStatusGetResponseTypeMap.put(400, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalCheckStatusGetResponseTypeMap.put(401, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalCheckStatusGetResponseTypeMap.put(403, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalCheckStatusGetResponseTypeMap.put(500, new GenericType<StreetAccountStatus>(){});
  }

  private static final Map<Integer, GenericType> streetaccountHistoricalGetFilesGetResponseTypeMap = new HashMap<Integer, GenericType>();
  static {
    streetaccountHistoricalGetFilesGetResponseTypeMap.put(200, new GenericType<GetfilesResponse>(){});
    streetaccountHistoricalGetFilesGetResponseTypeMap.put(400, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalGetFilesGetResponseTypeMap.put(401, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalGetFilesGetResponseTypeMap.put(403, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalGetFilesGetResponseTypeMap.put(500, new GenericType<StreetAccountStatus>(){});
  }

  private static final Map<Integer, GenericType> streetaccountHistoricalRequestFilesGetResponseTypeMap = new HashMap<Integer, GenericType>();
  static {
    streetaccountHistoricalRequestFilesGetResponseTypeMap.put(200, new GenericType<RequestfilesResponse>(){});
    streetaccountHistoricalRequestFilesGetResponseTypeMap.put(400, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalRequestFilesGetResponseTypeMap.put(401, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalRequestFilesGetResponseTypeMap.put(403, new GenericType<StreetAccountStatus>(){});
    streetaccountHistoricalRequestFilesGetResponseTypeMap.put(500, new GenericType<StreetAccountStatus>(){});
  }

  


  /**
   * Get the API client
   *
   * @return API client
   */
  public ApiClient getApiClient() {
    return apiClient;
  }

  /**
   * Set the API client
   *
   * @param apiClient an instance of API client
   */
  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  /**
   * Returns the status and percentage of completion for the requested jobID
   * Need to plug-in the jobID from /request-files into /check-status endpoint
   * @param jobID jobID returned by the request-files endpoint to know the status and percentDone (required)
   * @param paginationLimit Specifies the maximum number of results to return per result (optional)
   * @param paginationOffset Specifies the starting point for pagination. This parameter is used to identify the   beginning of next set of results (optional)
   * @return CheckstatusResponse
   * @throws ApiException if fails to make API call
   * @http.response.details
     <table summary="Response Details" border="1">
       <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
       <tr><td> 200 </td><td> Success </td><td>  -  </td></tr>
       <tr><td> 400 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 401 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 403 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 500 </td><td> Bad request </td><td>  -  </td></tr>
     </table>
   */
  public CheckstatusResponse streetaccountHistoricalCheckStatusGet(String jobID, Integer paginationLimit, Integer paginationOffset) throws ApiException {
    return streetaccountHistoricalCheckStatusGetWithHttpInfo(jobID, paginationLimit, paginationOffset).getData();
  }

  /**
   * Returns the status and percentage of completion for the requested jobID
   * Need to plug-in the jobID from /request-files into /check-status endpoint
   * @param jobID jobID returned by the request-files endpoint to know the status and percentDone (required)
   * @param paginationLimit Specifies the maximum number of results to return per result (optional)
   * @param paginationOffset Specifies the starting point for pagination. This parameter is used to identify the   beginning of next set of results (optional)
   * @return ApiResponse&lt;CheckstatusResponse&gt;
   * @throws ApiException if fails to make API call
   * @http.response.details
     <table summary="Response Details" border="1">
       <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
       <tr><td> 200 </td><td> Success </td><td>  -  </td></tr>
       <tr><td> 400 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 401 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 403 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 500 </td><td> Bad request </td><td>  -  </td></tr>
     </table>
   */
  public ApiResponse<CheckstatusResponse> streetaccountHistoricalCheckStatusGetWithHttpInfo(String jobID, Integer paginationLimit, Integer paginationOffset) throws ApiException {
    Object localVarPostBody = null;
    
    // verify the required parameter 'jobID' is set
    if (jobID == null) {
      throw new ApiException(400, "Missing the required parameter 'jobID' when calling streetaccountHistoricalCheckStatusGet");
    }
    
    // create path and map variables
    String localVarPath = "/streetaccount/historical/check-status";

    // query params
    java.util.List<Pair> localVarQueryParams = new java.util.ArrayList<Pair>();
    java.util.Map<String, String> localVarHeaderParams = new java.util.HashMap<String, String>();
    java.util.Map<String, String> localVarCookieParams = new java.util.HashMap<String, String>();
    java.util.Map<String, Object> localVarFormParams = new java.util.HashMap<String, Object>();

    localVarQueryParams.addAll(apiClient.parameterToPairs("", "jobID", jobID));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "_paginationLimit", paginationLimit));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "_paginationOffset", paginationOffset));

    
    
    
    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "FactSetApiKey", "FactSetOAuth2", "FactSetOAuth2Client" };


    ApiResponse<
        
        CheckstatusResponse
      
    > apiResponse = apiClient.invokeAPI("StreetAccountHistoricalStoriesApi.streetaccountHistoricalCheckStatusGet", localVarPath, "GET", localVarQueryParams, localVarPostBody,
                               localVarHeaderParams, localVarCookieParams, localVarFormParams, localVarAccept, localVarContentType,
                               localVarAuthNames, streetaccountHistoricalCheckStatusGetResponseTypeMap, false);

    return apiResponse;

  }
  /**
   * Returns the StreetAccount XML files for the specified date range
   * Need to plug-in the jobID from /check-status into /get-files endpoint
   * @param jobID jobID returned by the request-files endpoint to collect the results of the query (required)
   * @param paginationLimit Specifies the maximum number of results to return per result (optional)
   * @param paginationOffset Specifies the starting point for pagination. This parameter is used to identify the   beginning of next set of results (optional)
   * @return GetfilesResponse
   * @throws ApiException if fails to make API call
   * @http.response.details
     <table summary="Response Details" border="1">
       <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
       <tr><td> 200 </td><td> success </td><td>  -  </td></tr>
       <tr><td> 400 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 401 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 403 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 500 </td><td> Bad request </td><td>  -  </td></tr>
     </table>
   */
  public GetfilesResponse streetaccountHistoricalGetFilesGet(String jobID, Integer paginationLimit, Integer paginationOffset) throws ApiException {
    return streetaccountHistoricalGetFilesGetWithHttpInfo(jobID, paginationLimit, paginationOffset).getData();
  }

  /**
   * Returns the StreetAccount XML files for the specified date range
   * Need to plug-in the jobID from /check-status into /get-files endpoint
   * @param jobID jobID returned by the request-files endpoint to collect the results of the query (required)
   * @param paginationLimit Specifies the maximum number of results to return per result (optional)
   * @param paginationOffset Specifies the starting point for pagination. This parameter is used to identify the   beginning of next set of results (optional)
   * @return ApiResponse&lt;GetfilesResponse&gt;
   * @throws ApiException if fails to make API call
   * @http.response.details
     <table summary="Response Details" border="1">
       <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
       <tr><td> 200 </td><td> success </td><td>  -  </td></tr>
       <tr><td> 400 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 401 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 403 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 500 </td><td> Bad request </td><td>  -  </td></tr>
     </table>
   */
  public ApiResponse<GetfilesResponse> streetaccountHistoricalGetFilesGetWithHttpInfo(String jobID, Integer paginationLimit, Integer paginationOffset) throws ApiException {
    Object localVarPostBody = null;
    
    // verify the required parameter 'jobID' is set
    if (jobID == null) {
      throw new ApiException(400, "Missing the required parameter 'jobID' when calling streetaccountHistoricalGetFilesGet");
    }
    
    // create path and map variables
    String localVarPath = "/streetaccount/historical/get-files";

    // query params
    java.util.List<Pair> localVarQueryParams = new java.util.ArrayList<Pair>();
    java.util.Map<String, String> localVarHeaderParams = new java.util.HashMap<String, String>();
    java.util.Map<String, String> localVarCookieParams = new java.util.HashMap<String, String>();
    java.util.Map<String, Object> localVarFormParams = new java.util.HashMap<String, Object>();

    localVarQueryParams.addAll(apiClient.parameterToPairs("", "jobID", jobID));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "_paginationLimit", paginationLimit));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "_paginationOffset", paginationOffset));

    
    
    
    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "FactSetApiKey", "FactSetOAuth2", "FactSetOAuth2Client" };


    ApiResponse<
        
        GetfilesResponse
      
    > apiResponse = apiClient.invokeAPI("StreetAccountHistoricalStoriesApi.streetaccountHistoricalGetFilesGet", localVarPath, "GET", localVarQueryParams, localVarPostBody,
                               localVarHeaderParams, localVarCookieParams, localVarFormParams, localVarAccept, localVarContentType,
                               localVarAuthNames, streetaccountHistoricalGetFilesGetResponseTypeMap, false);

    return apiResponse;

  }
  /**
   * Returns the jobID
   * Give the startDate and endDate parameters as request parameters in the /request-files endpoint, it returns the jobID. startDate and endDate should be in YYYY-MM-DDTHH:MM:SSZ format. This API only supports adhoc requests to retrieve historical files and does not support real-time files and if you interested in require real-time push should consider the other three methods (pushed via SFTP, to QNT account, or your Azure Storage). Per API request able to query till 2 years of data
   * @param startDate Date from which data is required. Should be YYYY-MM-DDTHH:MM:SSZ format (required)
   * @param endDate The date until which the data is to be fetched. Should be YYYY-MM-DDTHH:MM:SSZ format (required)
   * @return RequestfilesResponse
   * @throws ApiException if fails to make API call
   * @http.response.details
     <table summary="Response Details" border="1">
       <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
       <tr><td> 200 </td><td> success </td><td>  -  </td></tr>
       <tr><td> 400 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 401 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 403 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 500 </td><td> Bad request </td><td>  -  </td></tr>
     </table>
   */
  public RequestfilesResponse streetaccountHistoricalRequestFilesGet(OffsetDateTime startDate, OffsetDateTime endDate) throws ApiException {
    return streetaccountHistoricalRequestFilesGetWithHttpInfo(startDate, endDate).getData();
  }

  /**
   * Returns the jobID
   * Give the startDate and endDate parameters as request parameters in the /request-files endpoint, it returns the jobID. startDate and endDate should be in YYYY-MM-DDTHH:MM:SSZ format. This API only supports adhoc requests to retrieve historical files and does not support real-time files and if you interested in require real-time push should consider the other three methods (pushed via SFTP, to QNT account, or your Azure Storage). Per API request able to query till 2 years of data
   * @param startDate Date from which data is required. Should be YYYY-MM-DDTHH:MM:SSZ format (required)
   * @param endDate The date until which the data is to be fetched. Should be YYYY-MM-DDTHH:MM:SSZ format (required)
   * @return ApiResponse&lt;RequestfilesResponse&gt;
   * @throws ApiException if fails to make API call
   * @http.response.details
     <table summary="Response Details" border="1">
       <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
       <tr><td> 200 </td><td> success </td><td>  -  </td></tr>
       <tr><td> 400 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 401 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 403 </td><td> Bad request </td><td>  -  </td></tr>
       <tr><td> 500 </td><td> Bad request </td><td>  -  </td></tr>
     </table>
   */
  public ApiResponse<RequestfilesResponse> streetaccountHistoricalRequestFilesGetWithHttpInfo(OffsetDateTime startDate, OffsetDateTime endDate) throws ApiException {
    Object localVarPostBody = null;
    
    // verify the required parameter 'startDate' is set
    if (startDate == null) {
      throw new ApiException(400, "Missing the required parameter 'startDate' when calling streetaccountHistoricalRequestFilesGet");
    }
    
    // verify the required parameter 'endDate' is set
    if (endDate == null) {
      throw new ApiException(400, "Missing the required parameter 'endDate' when calling streetaccountHistoricalRequestFilesGet");
    }
    
    // create path and map variables
    String localVarPath = "/streetaccount/historical/request-files";

    // query params
    java.util.List<Pair> localVarQueryParams = new java.util.ArrayList<Pair>();
    java.util.Map<String, String> localVarHeaderParams = new java.util.HashMap<String, String>();
    java.util.Map<String, String> localVarCookieParams = new java.util.HashMap<String, String>();
    java.util.Map<String, Object> localVarFormParams = new java.util.HashMap<String, Object>();

    localVarQueryParams.addAll(apiClient.parameterToPairs("", "startDate", startDate));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "endDate", endDate));

    
    
    
    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "FactSetApiKey", "FactSetOAuth2", "FactSetOAuth2Client" };


    ApiResponse<
        
        RequestfilesResponse
      
    > apiResponse = apiClient.invokeAPI("StreetAccountHistoricalStoriesApi.streetaccountHistoricalRequestFilesGet", localVarPath, "GET", localVarQueryParams, localVarPostBody,
                               localVarHeaderParams, localVarCookieParams, localVarFormParams, localVarAccept, localVarContentType,
                               localVarAuthNames, streetaccountHistoricalRequestFilesGetResponseTypeMap, false);

    return apiResponse;

  }
}

File: code/typescript/StreetAccountNews/v0/src/api/StreetAccountHistoricalStoriesApi.js
/**
 * StreetAccount News API
 * The StreetAccount News API provides access to FactSet's proprietary news provider, StreetAccount. StreetAccount, is a premium real-time market intelligence news service that delivers comprehensive U.S., Canadian, and European coverage (and expanding Asia coverage). All possible sources for corporate news are scanned and key story facts are highlighted and presented in an easy-to-read format.  **StreetAccount Filters, Headlines, and Views:**  These endpoints allow for the retrieval of news headlines using filters such as Watchlists/Indices/Tickers, Company Filters, Market Topics, Regions, and Sectors.  Headlines can also be retrieved based on saved views within the Workstation.  **StreetAccount Historical Stories:**  These endpoints provide access to historical StreetAccount (SA) news. The API delivers SA stories in XML format based on user-specified date input parameters. Output files are securely available to users through a URL. This API has three endpoints: Request Files, Check Status, and Get Files.  Please note that this API only supports adhoc requests for historical files and does not support real-time files. If real-time push is required, consider using other methods such as SFTP, QNT account, or Azure Storage. Both historical and real-time Street Account news can be delivered via SFTP, QNT account, or Azure Storage.  The files delivered contain both metadata and content body, eliminating the need for multiple requests through multiple services to retrieve all the information.
 *
 * The version of the OpenAPI document: 0.1.0
 * Contact: api@factset.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */


import ApiClient from "../ApiClient";
import CheckstatusResponse from '../model/CheckstatusResponse';
import GetfilesResponse from '../model/GetfilesResponse';
import RequestfilesResponse from '../model/RequestfilesResponse';
import StreetAccountStatus from '../model/StreetAccountStatus';

/**
* StreetAccountHistoricalStories service.
* @module api/StreetAccountHistoricalStoriesApi
*/
export default class StreetAccountHistoricalStoriesApi {

    /**
    * Constructs a new StreetAccountHistoricalStoriesApi. 
    * @alias module:api/StreetAccountHistoricalStoriesApi
    * @class
    * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
    * default to {@link module:ApiClient#instance} if unspecified.
    */
    constructor(apiClient) {
        this.apiClient = apiClient || ApiClient.instance;
    }



    /**
     * Returns the status and percentage of completion for the requested jobID
     * Need to plug-in the jobID from /request-files into /check-status endpoint
     * @param {String} jobID jobID returned by the request-files endpoint to know the status and percentDone
     * @param {Object} opts Optional parameters
     * @param {Number} opts.paginationLimit Specifies the maximum number of results to return per result
     * @param {Number} opts.paginationOffset Specifies the starting point for pagination. This parameter is used to identify the   beginning of next set of results
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/CheckstatusResponse} and HTTP response
     */
    streetaccountHistoricalCheckStatusGetWithHttpInfo(jobID, opts) {
      opts = opts || {};
      let postBody = null;
      // verify the required parameter 'jobID' is set
      if (jobID === undefined || jobID === null) {
        throw new Error("Missing the required parameter 'jobID' when calling streetaccountHistoricalCheckStatusGet");
      }

      let pathParams = {
      };
      let queryParams = {
        'jobID': jobID,
        '_paginationLimit': opts['paginationLimit'],
        '_paginationOffset': opts['paginationOffset']
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['FactSetApiKey', 'FactSetOAuth2'];
      let contentTypes = [];
      let accepts = ['application/json'];


      let returnType = CheckstatusResponse;

      return this.apiClient.callApi(
        '/streetaccount/historical/check-status', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null
      );
    }

    /**
     * Returns the status and percentage of completion for the requested jobID
     * Need to plug-in the jobID from /request-files into /check-status endpoint
     * @param {String} jobID jobID returned by the request-files endpoint to know the status and percentDone
     * @param {Object} opts Optional parameters
     * @param {Number} opts.paginationLimit Specifies the maximum number of results to return per result
     * @param {Number} opts.paginationOffset Specifies the starting point for pagination. This parameter is used to identify the   beginning of next set of results
     * @return { Promise.< module:model/CheckstatusResponse > } a Promise, with data of type {@link module:model/CheckstatusResponse }
     */
    streetaccountHistoricalCheckStatusGet(jobID, opts) {
      return this.streetaccountHistoricalCheckStatusGetWithHttpInfo(jobID, opts)
        .then(function(response_and_data) {
          return response_and_data.data;
        });
    }


    /**
     * Returns the StreetAccount XML files for the specified date range
     * Need to plug-in the jobID from /check-status into /get-files endpoint
     * @param {String} jobID jobID returned by the request-files endpoint to collect the results of the query
     * @param {Object} opts Optional parameters
     * @param {Number} opts.paginationLimit Specifies the maximum number of results to return per result
     * @param {Number} opts.paginationOffset Specifies the starting point for pagination. This parameter is used to identify the   beginning of next set of results
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/GetfilesResponse} and HTTP response
     */
    streetaccountHistoricalGetFilesGetWithHttpInfo(jobID, opts) {
      opts = opts || {};
      let postBody = null;
      // verify the required parameter 'jobID' is set
      if (jobID === undefined || jobID === null) {
        throw new Error("Missing the required parameter 'jobID' when calling streetaccountHistoricalGetFilesGet");
      }

      let pathParams = {
      };
      let queryParams = {
        'jobID': jobID,
        '_paginationLimit': opts['paginationLimit'],
        '_paginationOffset': opts['paginationOffset']
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['FactSetApiKey', 'FactSetOAuth2'];
      let contentTypes = [];
      let accepts = ['application/json'];


      let returnType = GetfilesResponse;

      return this.apiClient.callApi(
        '/streetaccount/historical/get-files', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null
      );
    }

    /**
     * Returns the StreetAccount XML files for the specified date range
     * Need to plug-in the jobID from /check-status into /get-files endpoint
     * @param {String} jobID jobID returned by the request-files endpoint to collect the results of the query
     * @param {Object} opts Optional parameters
     * @param {Number} opts.paginationLimit Specifies the maximum number of results to return per result
     * @param {Number} opts.paginationOffset Specifies the starting point for pagination. This parameter is used to identify the   beginning of next set of results
     * @return { Promise.< module:model/GetfilesResponse > } a Promise, with data of type {@link module:model/GetfilesResponse }
     */
    streetaccountHistoricalGetFilesGet(jobID, opts) {
      return this.streetaccountHistoricalGetFilesGetWithHttpInfo(jobID, opts)
        .then(function(response_and_data) {
          return response_and_data.data;
        });
    }


    /**
     * Returns the jobID
     * Give the startDate and endDate parameters as request parameters in the /request-files endpoint, it returns the jobID. startDate and endDate should be in YYYY-MM-DDTHH:MM:SSZ format. This API only supports adhoc requests to retrieve historical files and does not support real-time files and if you interested in require real-time push should consider the other three methods (pushed via SFTP, to QNT account, or your Azure Storage). Per API request able to query till 2 years of data
     * @param {Date} startDate Date from which data is required. Should be YYYY-MM-DDTHH:MM:SSZ format
     * @param {Date} endDate The date until which the data is to be fetched. Should be YYYY-MM-DDTHH:MM:SSZ format
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/RequestfilesResponse} and HTTP response
     */
    streetaccountHistoricalRequestFilesGetWithHttpInfo(startDate, endDate) {
      let postBody = null;
      // verify the required parameter 'startDate' is set
      if (startDate === undefined || startDate === null) {
        throw new Error("Missing the required parameter 'startDate' when calling streetaccountHistoricalRequestFilesGet");
      }
      // verify the required parameter 'endDate' is set
      if (endDate === undefined || endDate === null) {
        throw new Error("Missing the required parameter 'endDate' when calling streetaccountHistoricalRequestFilesGet");
      }

      let pathParams = {
      };
      let queryParams = {
        'startDate': startDate,
        'endDate': endDate
      };
      let headerParams = {
      };
      let formParams = {
      };

      let authNames = ['FactSetApiKey', 'FactSetOAuth2'];
      let contentTypes = [];
      let accepts = ['application/json'];


      let returnType = RequestfilesResponse;

      return this.apiClient.callApi(
        '/streetaccount/historical/request-files', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, null
      );
    }

    /**
     * Returns the jobID
     * Give the startDate and endDate parameters as request parameters in the /request-files endpoint, it returns the jobID. startDate and endDate should be in YYYY-MM-DDTHH:MM:SSZ format. This API only supports adhoc requests to retrieve historical files and does not support real-time files and if you interested in require real-time push should consider the other three methods (pushed via SFTP, to QNT account, or your Azure Storage). Per API request able to query till 2 years of data
     * @param {Date} startDate Date from which data is required. Should be YYYY-MM-DDTHH:MM:SSZ format
     * @param {Date} endDate The date until which the data is to be fetched. Should be YYYY-MM-DDTHH:MM:SSZ format
     * @return { Promise.< module:model/RequestfilesResponse > } a Promise, with data of type {@link module:model/RequestfilesResponse }
     */
    streetaccountHistoricalRequestFilesGet(startDate, endDate) {
      return this.streetaccountHistoricalRequestFilesGetWithHttpInfo(startDate, endDate)
        .then(function(response_and_data) {
          return response_and_data.data;
        });
    }


}





